docs/2.8.0/docs/app-dev/bindings-java/open-tracing.rst

```diff
@@ -0,0 +1,170 @@
+
+The next step is to initialize the GrpcTelemetry controller, which is responsible for propagation of the trace contexts inside the HTTP2 headers of the Grpc communication.
+
+Conveniently, the example wraps the necessary initialization steps in the constructor of the OpenTelemetryUtil class. All you have to do is call
+
+.. code-block:: java
+
+    OpenTelemetryUtil openTelemetry = new OpenTelemetryUtil(APP_ID);
+
+The GrpcTelemetry controller can construct client call interceptors that need to be mounted on top of the **Netty** channels used in the grpc communication. The example provides a useful helper method called ``withClientInterceptor`` that injects an interceptor at channel builder level
+
+.. code-block:: java
+
+    ManagedChannel channel =
+      openTelemetry.withClientInterceptor(
+        ManagedChannelBuilder
+          .forAddress(host, port)
+          .usePlaintext()
+      ).build();
+
+And with that, you are all set to start generating own spans, reporting them to the **Jaeger** server and also propagating them transparently to the **Ledger API**.
+
+Starting New Spans
+==================
+
+Before making a Grpc call, you must generate a new span that will cover the multi-component interaction that is about to be initiated. The example provides a useful combinator called ``runInNewSpan`` that wraps the execution of an arbitrary function in a newly generated span
+
+.. code-block:: java
+
+    public <R> R runInNewSpan(String spanName, Supplier<R> body) {
+        Span span = tracer.spanBuilder(spanName).startSpan();
+        try(Scope ignored = span.makeCurrent()) {
+            return body.get();
+        } finally {
+            span.end();
+        }
+    }
+
+You can use it on a command submission in the following manner
+
+.. code-block:: java
+
+    openTelemetry.runInNewSpan("createInitialContracts", () -> submissionService.submit(request));
+
+The Grpc interceptors that have been mounted at the initialization stage, will do the rest of the work behind the scenes making sure that the spans make it across to the Canton.
+
+Continuing Spans Across Different Applications
+==============================================
+
+Sometimes, you may wish to continue the same span across multiple daml transactions forming a single workflow. This may be especially interesting when different client application instances interact through the ledger and yet their entire conversation should be seen as a single coherent succession of spans. In that case, it is possible to extract the trace context associated with the past transactions from the Transaction, TransactionTree or Completion records that are returned from the following **Ledger API** calls
```

Comment:
```suggestion
Sometimes you may wish to continue the same span across multiple Daml transactions forming a single workflow. This may be especially interesting when different client application instances interact through the ledger and yet their entire conversation should be seen as a single coherent succession of spans. In that case, it is possible to extract the trace context associated with the past transactions from the Transaction, TransactionTree, or Completion records that are returned from the following **Ledger API** calls:
```